src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
+{
+    public class Http3ConnectionTests : Http3TestBase
+    {
+        [Fact]
+        public async Task VerifySettingsAreReceived()
+        {
+            await InitializeConnectionAsync(_noopApplication);
+            await CreateOutboundControlStream(ControlStreamId);
+            await CreateOutboundControlStream(EncoderStreamId);
+            await CreateOutboundControlStream(DecoderStreamId);
+            await WaitForInboundControlStreamCreated();
+
+            var settings = await ReadSettings();
+            Assert.Equal(new KestrelServerLimits().MaxRequestHeadersTotalSize, settings[(long)Http3SettingType.MaxHeaderListSize]);
+        }
+
+        [Fact]
+        [Flaky("<No longer needed; tracked in Kusto>", FlakyOn.All)]
```

Comment:
I originally had a tcs to await which would only be updated after the max request payload size is updated. However, I thought that because we were using PipeScheduler.Inline, the call to WriteSettings would always update settings inline. I've observed this when stepping through the debugger as well.

I'm going to revert back to using a tcs and see if that helps with flakiness. I'm also going to leave this marked as flaky if this wasn't the actual issue.

Reply:
FYI, FlakyAttribute has been renamed to QuarantinedTestAttribute.